### PR TITLE
[feat] Create forgot password flow

### LINKED ISSUE #75: Create forgot password flow
See Linear for sprint description:
https://linear.app/trap-garden/issue/TG-51/create-forgot-password-flow

### PR DESCRIPTION
**Pull request by rachaelch3n**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Added reset your password flow where user can access page by clicking "Reset here". 
- User should receive email from supabase with instructions to reset password after inputing email and clicking send.
- User can only click send if valid email (structure) is in text input box.
- User will be pushed to email sent page regardless of whether email account is actually registered in supabase.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into
 the placeholder."

[image placeholder]: 

Screen Shot 2025-03-02 at 3 41 02 PM
Screen Shot 2025-03-02 at 3 40 40 PM
Screen Shot 2025-03-02 at 3 40 26 PM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Edited login page, created email-sent and forgot-password page, and lib/icons (for the back arrow icon). Confirm that you only receive an email from supabase if your account is registered. 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Might include message on email-sent page that states if the user didn't receive an email from supabse, then either the account isn't registered or the user input the email wrong.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: app/(auth)/forgot-password/page.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  const handleEmailChange = async (newEmail: string) => {
+    setEmail(newEmail);
+    setInvalidEmailError('');
+    setCheckValidEmailError(
+      !isValidEmail(newEmail) ? 'Please enter a valid email address' : '',
+    );
+  };
+
+  const isValidEmail = (email: string): boolean => {
```
Comment (kaleenchen) [inline app/(auth)/forgot-password/page.tsx]: isValidEmail can be outside component because not needed for component state / dont need to render each time

File: app/(auth)/email-sent/page.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use client';
```
Comment (kaleenchen) [inline app/(auth)/email-sent/page.tsx:1]: yay!! everything honestly looks good! <3

**Review by kaleenchen — APPROVED** [review]
looks good!

File: app/(auth)/email-sent/page.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import { useRouter } from 'next/navigation';
+import { BackButton } from '@/app/plant-page/style';
```
Comment (kylezryr) [inline app/(auth)/email-sent/page.tsx]: instead of importing BackButton from plant-page, could you copy over the code into the `auth/styles.ts` and import it from there instead? just in case the styling is different in the future

File: app/(auth)/forgot-password/page.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import Icon from '@/components/Icon';
+import TextInput from '@/components/TextInput';
+import COLORS from '@/styles/colors';
+import { H2, P3 } from '@/styles/text';
+import { StyledForm } from '../styles';
+
+const isValidEmail = (email: string): boolean => {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailRegex.test(email);
+};
```
Comment (kylezryr) [inline app/(auth)/forgot-password/page.tsx]: do you think you could move this `isValidEmail` function into `/utils/helpers.ts`, then import it here and also in `/signup`? since we're also using it in signup

File: app/(auth)/forgot-password/page.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  const [checkValidEmailError, setCheckValidEmailError] = useState('');
+  const canSubmitForm = email && !checkValidEmailError;
+
+  const router = useRouter();
+
+  const handleEmailChange = async (newEmail: string) => {
+    setEmail(newEmail);
+    setInvalidEmailError('');
+    setCheckValidEmailError(
+      !isValidEmail(newEmail) ? 'Please enter a valid email address' : '',
```
Comment (kylezryr) [inline app/(auth)/forgot-password/page.tsx]: i don't see this error message appear when typing in the email, if we want it to appear as the user types (which i think it should), we can put the `setCheckValidEmailError` section in a useEffect with a dependency of `newEmail`, so that it updates every time the user types in the text box

File: app/(auth)/email-sent/page.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+          router.push('/forgot-password');
+        }}
+      >
+        <Icon type={'backArrow'} />
+      </BackButton>
+
+      <H2 $color={COLORS.shrub} style={{ marginBottom: '8px' }}>
+        Email Sent!
+      </H2>
+      <div style={{ display: 'flex', flexDirection: 'column', gap: '1.5rem' }}>
```
Comment (kylezryr) [inline app/(auth)/email-sent/page.tsx]: you can replace this with a `<Flex>` container imported from `/styles/containers.ts` and assign the corresponding props for `flex-direction` and `gap`

File: app/(auth)/login/page.tsx [hunk 7]
```diff
@@ -101,6 +101,12 @@ export default function Login() {
           Log In
         </BigButton>
         {/* Sign in button */}
+        <P3 as="span" $color={COLORS.midgray} style={{ textAlign: 'center' }}>
+          Forgot Password?
+          <StyledLinkButton href="/forgot-password" style={{ padding: '4px' }}>
```
Comment (kylezryr) [inline app/(auth)/login/page.tsx]: instead of assigning in-line styles here, you can import these into the `styles.ts` files and create new styled components by doing this:

```
//import StyledLinkButton

export const ResetLinkButton = styled(StyledLinkButton)` 
    padding: 4px;
`

```

you can do this for both the P3 and StyledLinkButton, and anywhere else you may have done this, just keeps the code cleaner!

File: app/(auth)/forgot-password/page.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+          router.push('/login');
+        }}
+      >
+        <Icon type={'backArrow'} />
+      </BackButton>
+
+      <H2 $color={COLORS.shrub} style={{ marginBottom: '8px' }}>
+        Reset your password
+      </H2>
+      <div style={{ display: 'flex', flexDirection: 'column', gap: '1.5rem' }}>
```
Comment (kylezryr) [inline app/(auth)/forgot-password/page.tsx]: This can also be a `<Flex>` container

**Comment by kylezryr** [comment]
everything looks great, great job!! just a couple things:

1. you can replace divs with `Flex` containers 
2. try to replace in-line styling with creating new styled components of existing components 
3. move isValidEmail to `/utils/helpers.ts` and import it to use in sign up

**Review by kylezryr — APPROVED** [review]
everything looks good and ready to merge!
